server/bin/www

```diff
@@ -33,17 +37,47 @@ var server = http.createServer(app);
 //server.listen(port);
 //server.on("error", onError);
 //server.on("listening", onListening);
-models.sequelize.sync().then(function () {
+models.sequelize.sync().then(function() {
   /**
    * Listen on provided port, on all network interfaces.
    */
-  server.listen(port, function () {
+  server.listen(port, function() {
     debug("Express server listening on port " + server.address().port);
   });
   server.on("error", onError);
   server.on("listening", onListening);
 });
 
+/**
+ * Socket setup
+ */
+const io = socketio(server);
+
+io.on("connection", socket => {
+  console.log("We have a new connection.");
+  let encodedCookie = socket.handshake.headers.cookie;
+  var parsedCookies = cookie.parse(encodedCookie);
+
+  // User data of current Socket User stored in userData
+  // User data includes: userName, userEmail and userId
+  var userData = authJWTCheck(parsedCookies);
+
+  socket.on("getNotifications", async (message, callback) => {
+    // userNotifications gets all the notifications for a certain user
+    // userNotifications.dataValues = all the data values
+    // dataValues are: productId, userId, previousPrice, currentPrice, dismissed
+    let userNotifications = await notificationHandler.getProductsForNotifications(
+      userData.userId
+    );
+
+    callback(userNotifications);
+  });
+
+  socket.on("disconnect", () => {
+    console.log("User has left.");
+  });
+});
+
```

Comment:
For notifying clients, you want to add a method that takes the notifications and a userId. It should be used by the cronJob E.g:
```js
const sendNotification = (notifications, userId) =>{
const socketId = socketClients[userId]
if (!socketId) return 
 io.to(`${socketId}`).emit("notificationsEvent", notifications);

}
```

Then on the **client** side, the socket will listen to that particular event. E.g

```js
 socket.on("notificationsEvent", notifications => {
    //update state from the notifications
    });
```